crt_portal/cts_forms/views.py

```diff
@@ -182,6 +183,38 @@ def _related_reports_count(report):
         return 1
 
 
+def _get_email(report):
+    return report.contact_email
+
+
+def _set_constant_writer(contact_templates, report):
+    for contact_template in contact_templates:
+        if contact_template[1].upper() == report.contact_email.upper():
+            report.constant_writer = True
+
+
+def _was_sent_constant_writer_email(reports):
+    report_emails = tuple(map(_get_email, reports))
+    if len(report_emails) == 0:
+        return
+    with connection.cursor() as cursor:
+        try:
+            cursor.execute(
+                """  select act.description as description, r.contact_email as email
+                from cts_forms_report r
+                left join actstream_action act on concat('Email sent: ''CRT - Constant Writer'' to ', r.contact_email, ' via govDelivery TMS') = act.description
+                WHERE act.verb = 'Contacted complainant:'
+                AND r.contact_email IN %s
+                order by act.timestamp desc
+                """, params=[report_emails])
```

Comment:
FYI @ilodidoj 

Katie and I paired on this query and proved to ourselves this isn't possible with the ORM. Even if we were to construct a view, we'd need to use raw SQL to construct the view, and the data would be 24 hours delayed (where here the constant writer icon will appear instantly with this code)

- To get Django to construct a join, you need a ForeignKey between the two join targets.
- The ForeignKey must be based off of an existing field on the table.
- This join's "foreign key" is a constructed field (built with CONCAT)
- Django doesn't allow constructed fields like this to be defined at the model level
- So, we can't define a ForeignKey based on a constructed field